### DISCUSSION TITLE
USPS Commercial Rates and Cubic

### DISCUSSION THREAD

**Discussion by owi2000** [alexeybusygin/ShippingRates#54] [Q&A] [answered]
I'm able to get rates back from the USPS api, however the rates are only the retail rates.  Additionally, even though package dimensions are sent with the request, no cubic rates (i.e. Priority Mail cubic, or Ground Advantage Cubic) are returned.  How can I get these rates?

**Comment by alexeybusygin** [comment, marked as answer]
Cubic rates should be enabled for your account on the USPS side. You can try to fetch them by passing the cubic method name to the USPS:

`rateManager.AddProvider(new USPSProvider(uspsUserId, "PRIORITY MAIL CUBIC"));`

And setting appropriate container type:

```
var packages = new List<Package>
{
    new Package(12, 12, 12, 35, 150, "CUBIC PARCELS")
}
```

But if rates are not enabled for your account, then you will get this error in the response:
The requested Mail Service is not available for the specified customer type.
